### PR TITLE
Remove XBAP-related dead code from Application/BrowserInteropHelper, finish cleanup

### PR DESCRIPTION
**Pull request by h3xds1nz**
## Description

This continues the work started in #9855, removing rest of the internal properties in `BrowserInteropHelper` and thanks to this we can remove dead code in other places of the codebase (`Application`, `AppSecurityManager`, etc.) I've separated the two as those deletions might be a bit harder to grasp, while the first one is easier to understand imho.

- `SiteOfOriginContainer.BrowserSource` is never set, thus it is always `NULL`.
     - This means the public property `Uri` in `BrowserInteropHelper` in always `NULL`.
     - `GetReferer` method in `BindUriHelper` depends on this value, if `BrowserSource` is NULL, it returns `NULL`.
     - So the value in `httpRequest.Referer` of `WpfWebRequestHelper` would end up being `NULL` anyways, no-op / dead code.
- `Application.MimeType` is never set, thus it is always `MimeType.Unknown`.
     - This means `IsViewer` of `BrowserInteropHelper` is always `false`.
     - That also means that `IsInitialViewerNavigation` is always `false` and its underlying value does not matter at all for any operations. So we can remove any setters; and then we can remove it from both conditions; which makes it unused.
- `Application.ServiceProvider` is never set (classes doing it don't exist anymore), so it is always `NULL`.
- Classes calling the `GetService` also don't exist anymore, so the whole method does not matter.
- Any methods removed in `AppSecurityManager` just call themselves, besides that we were still calling `ClearSecurityManager` from Application but it is never initialized anymore so there's nothing to clear.
- As a result, `SecurityMgrSite` is never initialized anymore, so we can remove it as well.

## Customer Impact

Decreased size of assemblies, tiny bit of perf, cleaner codebase for developers.

## Regression

No.

## Testing

Local build, app run.

## Risk

Low, the code is a dead code or it doesn't do anything.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9865)

### CODE REVIEW

**Comment by h3xds1nz** [comment]
Looks like one of the changes in #9220 was here. Rebased and resolved merge conflicts.

**Comment by h3xds1nz** [comment]
Resolved merge conflicts #01

**Comment by h3xds1nz** [comment]
Fix merge conflicts after #9521

**Comment by h3xds1nz** [comment]
Resolved merge conflicts after #9855 

@harshit7962 Can we go with this one and #9943 next?

**Comment by harshit7962** [comment]
@h3xds1nz, We are done with the test pass of #9943 and will prioritize it. I am yet to review this PR and will include it in the next test run.

**Comment by h3xds1nz** [comment]
@harshit7962 Awesome, let me know at which part you'd like me to resolve the conflicts.

I feel its counter-productive to do it now should a style-PR get merged meanwhile as that will mess up the resolutions.

**Comment by harshit7962** [comment]
Sure, will let you know when the PR is ready to be merged. It would be better to resolve the conflicts at that time.

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/BrowserInteropHelper.cs [hunk 1]
```diff
@@ -41,53 +32,13 @@ static BrowserInteropHelper()
         /// <summary>
         /// Returns true if the app is a browser hosted app.
         /// </summary>
-        /// <remarks>
```
Comment (harshit7962) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/BrowserInteropHelper.cs:44]: Was this intentionally removed?

Comment (h3xds1nz) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/BrowserInteropHelper.cs:44]: Yeah, `HostingFlags` was an internal static property that is no longer present since #9855, just forgot to clean up before.

**Review by harshit7962 — APPROVED** [review]

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/BrowserInteropHelper.cs (hunk 1 above)
Comment (harshit7962) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/BrowserInteropHelper.cs:44]: Yeah makes sense. Other changes seems fine. Will speed up the test pass and try to get this in.

**Comment by harshit7962** [comment]
@h3xds1nz, we are done with the test pass and its all green. We are ready to take this change in.

**Comment by h3xds1nz** [comment]
> @h3xds1nz, we are done with the test pass and its all green. We are ready to take this change in.

@harshit7962 Great, should be merged/rebased properly I hope. I can't compile right now so I have to verify with CI I did pick the correct usings.

**Review by harshit7962 — APPROVED** [review]

**Comment by harshit7962** [comment]
@h3xds1nz, thanks for the contribution.

We are working on getting in the PRs that have passed the tests. Please let us know if any of these you would like us to prioritize.

Also, feel free to mention any PR that you would like us to include in the next test run.

**Comment by h3xds1nz** [comment]
@harshit7962 Great, glad to see this one in.

The untested ones in order of importance (perf gains):

1) #9364
2) #10003
3) #9949
4) #9199
5) #9395
6) #9844 (this is a big gain but the customer impact is lower as its more of an edge case)

Do note there are two unit test PRs that could be merged imho:

1) #9895
2) #9891 (this one includes a fix that slipped in through the WPF tests)

Regarding the ones that have passed tests so I can do further work/would be good to merge:

1) #9822
1.1) #9739
2) #9669
3) #9481 (@dipeshmsft was reviewing it)
4) #9468 (this provides a regression fix)
